lib/chai/core/assertions.js

```diff
@@ -333,22 +333,27 @@ module.exports = function (chai, _) {
 
   /**
    * ### .NaN
-   * Asserts that the target is `NaN`.
+   * Asserts that the target is a number and is exactly `NaN. No conversions are made.
+   * Relies on Number.isNaN() function or a polyfill
    *
-   *     expect('foo').to.be.NaN;
+   *     expect( Math.sqrt(-1) ).to.be.NaN;
    *     expect(4).not.to.be.NaN;
+   *     expect('foo').not.to.be.NaN;
    *
    * @name NaN
    * @api public
    */
-
-  Assertion.addProperty('NaN', function () {
-    this.assert(
-        isNaN(flag(this, 'object'))
-        , 'expected #{this} to be NaN'
-        , 'expected #{this} not to be NaN'
-    );
-  });
+  (function () {
+    // use standard Number.isNaN (ES6) or its polyfill (as proposed on MDN)
+    var isNanFunc = Number.isNaN || function (val) { return typeof val === 'number' && val !== val; };
```

Comment:
@thefourtheye 
You're right. I left the type check proposed on MDN in case we would be using ES6 Number.isNan(), but the [spec](http://www.ecma-international.org/ecma-262/6.0/#sec-number.isnan) for it states that it does a typecheck on its argument.

Are you OK with the following change on line 348
```javascript
var isNanFunc = Number.isNaN || function (val) { return val !== val; };
```